components/script/dom/htmlformelement.rs

```diff
@@ -465,6 +466,22 @@ impl HTMLFormElement {
     /// https://html.spec.whatwg.org/multipage/#statically-validate-the-constraints
     fn static_validation(&self) -> Result<(), Vec<FormSubmittableElement>> {
         let node = self.upcast::<Node>();
+        for child in node.traverse_preorder() {
+            if let Some(el) = child.downcast::<Element>() {
+                if el.disabled_state() {
+                    continue;
+                }
+                let validatable = match el.as_maybe_validatable() {
+                    Some(v) => v,
+                    None => continue,
+                };
+                if !validatable.is_instance_validatable() {
+                    continue;
+                }
+                // pass empty flags initially and set appropriate flags according to the type of element called
+                validatable.validate(ValidationFlags::empty());
+            }
+        }
```

Comment:
Hello Josh,

Trying to integrate the code, but every time it is giving some different kind of error on compilation.

Please correct me if my approach is wrong here:

This section of code https://github.com/servo/servo/blob/234951e4b6d66891cd0cc9705f42e3ca16d9e3d5/components/script/dom/htmlformelement.rs#L469

will traverse through the page and then the code which we wrote for calling validate method will be written in that section. if it returns true then None, else return Some(el). 

Now, I tried writing the code but I am struggling with rust specific errors here. 

Please guide me here.